custom/enikshay/tasks.py

```diff
@@ -156,17 +164,24 @@ def run_batch(self, case_ids):
         error_count = 0
         success_count = 0
         case_batches = 0
+        ledger_batches = 0
 
         errors = []
         with Timer() as t:
             batch_size = 100
-            updates = []
+            case_updates = []
+            ledger_updates = []
             for episode in self._get_open_episode_cases(case_ids):
                 did_error = False
                 update_json = {}
                 for updater in self.updaters:
                     try:
-                        potential_update = updater(self.domain, episode).update_json()
+                        _updater = updater(self.domain, episode)
+                        potential_update = _updater.update_json()
+                        if isinstance(_updater, EpisodeAdherenceUpdate):
```

Comment:
I still think you shouldn't have to reference the `EpisodeAdherenceUpdate` here at all by either checking if the updater that you are currently calling has the ledger_updates() method, or by adding that as a noop as the base class.

Reply:
Updated to
```
if hasattr(_updater, 'ledger_updates'):
    ledger_updates.extend(_updater.ledger_updates())
```

Unclear though why that makes it better?